services/galley/src/Galley/API/Meetings.hs

```diff
@@ -0,0 +1,235 @@
+import Wire.API.Error
+import Wire.API.Error.Galley
+import Wire.API.Federation.Client (FederatorClient)
+import Wire.API.Federation.Error
+import Wire.API.Meeting
+import Wire.API.Team.Feature (FeatureStatus (..), LockableFeature (..), MeetingConfig, MeetingPremiumConfig)
+import Wire.FederationAPIAccess ()
+import Wire.MeetingsSubsystem qualified as Meetings
+import Wire.NotificationSubsystem
+import Wire.Sem.Now (Now)
+import Wire.TeamStore qualified as TeamStore
+
+-- | Check if meetings feature is enabled for the user (if they're in a team)
+checkMeetingsEnabled ::
+  ( Member TeamStore r,
+    Member TeamFeatureStore r,
+    Member (ErrorS 'InvalidOperation) r,
+    Member (Input Opts) r
+  ) =>
+  UserId ->
+  Sem r ()
+checkMeetingsEnabled userId = do
+  maybeTeamId <- TeamStore.getOneUserTeam userId
+  case maybeTeamId of
+    Nothing -> pure () -- Personal users can use meetings
+    Just teamId -> do
+      meetingFeature <- getFeatureForTeam @MeetingConfig teamId
+      unless (meetingFeature.status == FeatureStatusEnabled) $
+        throwS @'InvalidOperation
+
+createMeeting ::
+  ( Member Meetings.MeetingsSubsystem r,
+    Member (ErrorS 'InvalidOperation) r,
+    Member (ErrorS 'NotATeamMember) r,
+    Member BackendNotificationQueueAccess r,
+    Member ConversationStore r,
+    Member (Error FederationError) r,
+    Member (Error InternalError) r,
+    Member (Error UnreachableBackends) r,
+    Member (FederationAPIAccess FederatorClient) r,
+    Member NotificationSubsystem r,
+    Member Now r,
+    Member P.TinyLog r,
+    Member TeamStore r,
+    Member TeamFeatureStore r,
+    Member (Input Opts) r
+  ) =>
+  Local UserId ->
+  NewMeeting ->
+  Sem r Meeting
+createMeeting lUser newMeeting = do
+  -- Check if meetings feature is enabled
+  checkMeetingsEnabled (tUnqualified lUser)
+
+  -- Validate that endDate > startDate
+  when (newMeeting.endDate <= newMeeting.startDate) $
+    throwS @'InvalidOperation
+
+  -- Determine trial status based on team membership and premium feature
+  maybeTeamId <- TeamStore.getOneUserTeam (tUnqualified lUser)
+  trial <- case maybeTeamId of
+    Nothing -> pure True -- Personal users create trial meetings
+    Just teamId -> do
+      -- Verify user is a team member (not just a collaborator)
+      maybeMember <- TeamStore.getTeamMember teamId (tUnqualified lUser)
+      case maybeMember of
+        Nothing -> throwS @'NotATeamMember -- User not a member
+        Just _member -> do
+          -- Check meeting premium feature status to determine trial
+          premiumFeature <- getFeatureForTeam @MeetingPremiumConfig teamId
+          pure $ case premiumFeature of
+            LockableFeature {status = FeatureStatusEnabled} -> False -- premium team, not trial
+            _ -> True -- non-premium team or disabled, is trial
+  (meeting, conversation) <- Meetings.createMeeting lUser newMeeting trial
+  notifyCreatedConversation lUser Nothing conversation InternalAdd
+  pure meeting
```

Comment:
I know, the current state does not suggest this, but we want all the business logic to be implemented in the subsystems. The code in the services should only be responsible for bootstrapping and wiring up the API. So all of this should go into the MeetingSubsystemInterpreter.